compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1945,6 +1945,29 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
                     );
                     (xform_self_ty, xform_ret_ty) =
                         self.xform_self_ty(probe.item, trait_ref.self_ty(), trait_ref.args);
+
+                    if matches!(probe.kind, WhereClauseCandidate(_)) {
```

Comment:
We could fast path this for the old trait solver, since we expect things to be normalized always.

Reply:
actually, move this `structurally_normalize_ty` above the `xform` and then use the normalized `ty` in `xform_self_ty`? This should avoid some work given that this change seems to actually impact perf in the new solver

Reply:
I don't see how moving it would change things for the better here, and it makes the structure kinda worse, so I'm not going to do this.